app/course-scraper/src/scrapers/google_cloud_skill_boost/scrape_journey.py

```diff
@@ -17,37 +17,57 @@
 
 
 # Open Journey Path
-def extract_ml_learning_path() -> list[dict]:
+def extract_ml_learning_path(GCSB_JOURNEY_URL) -> list[dict]:
+    # Send a request to the provided URL
     r = requests.get(GCSB_JOURNEY_URL)
     html_parser = etree.HTMLParser()
     dom = etree.fromstring(r.content, html_parser)
 
     data = []
     for journey in dom.xpath(pages.GCSBLearningJourneyPage.journeys):
+        try:
+            # Try to extract the first element from journey details
+            details = journey.xpath(pages.GCSBLearningJourneyPage.journey_details)[0]
+        except IndexError:
+            # If the first element is not available, use the full result or provide a default value
+            details = journey.xpath(pages.GCSBLearningJourneyPage.journey_details)
+            details = details if details else "No details available"
+
+        try:
+            # Try to extract the first link and construct the full URL
+            link = urljoin(GCSB_HOME_URL, journey.xpath(pages.GCSBLearningJourneyPage.journey_link)[0])
+        except IndexError:
+            # If the link is missing, handle it gracefully
+            link = urljoin(GCSB_HOME_URL, journey.xpath(pages.GCSBLearningJourneyPage.journey_link))
+            link = link if link else "No link available"
+
+        # Append the extracted information to the data list
         data.append(
             {
-                "title": journey.xpath(pages.GCSBLearningJourneyPage.journey_title)[0],
-                "details": journey.xpath(pages.GCSBLearningJourneyPage.journey_details)[
-                    0
-                ],
-                "description": journey.xpath(
-                    pages.GCSBLearningJourneyPage.journey_description
-                )[0],
-                "link": urljoin(
-                    GCSB_HOME_URL,
-                    journey.xpath(pages.GCSBLearningJourneyPage.journey_link)[0],
-                ),
+                "title": journey.xpath(pages.GCSBLearningJourneyPage.journey_title)[0] if journey.xpath(pages.GCSBLearningJourneyPage.journey_title) else "No title available",
+                "details": details,  # Use the result from the try-except block
+                "description": journey.xpath(pages.GCSBLearningJourneyPage.journey_description)[0] if journey.xpath(pages.GCSBLearningJourneyPage.journey_description) else "No description available",
+                "link": link,
             }
         )
 
     return data
 
+if __name__ == "__main__":
```

Comment:
Currently the GSCB_JOURNEY_URL is hard coded so it needs to manually be updated.  This process allows you to dynamically enter a different URL each time.

Reply:
I have updated the code, this will now first check if there is a GCSB_JOURNEY_URL in CONFIG, if so it will use that value but if not it will now prompt the user to enter the URL